### PR TITLE
Employ continuationToken for listArtifact

### PR DESCRIPTION
**Pull request by jonasfj**
I don't think this breaks anything....
Query string parameters are optional, so old client will just keep on working as they do now :)
Notice that I didn't have to modify the old tests, just add a new test case.

right now we list up to 10k without continuation token... After this we'll only be able to list up to 1k without continuation token... But with continuation token we'll be unbounded...

I doubt anyone has more than 1k artifacts anyways... So this should do fine :)

### CODE REVIEW

**Comment by jonasfj** [comment]
This will fix: https://bugzilla.mozilla.org/show_bug.cgi?id=1240189

File: routes/artifacts.js [hunk 1]
```diff
@@ -571,44 +568,45 @@ api.declare({
...
+    "you may limit this with the query-string parameter `limit`.",
   ].join('\n')
 }, async function(req ,res) {
-  var taskId = req.params.taskId;
-  var runId  = parseInt(req.params.runId);
+  let taskId        = req.params.taskId;
+  let runId         = parseInt(req.params.runId);
+  let continuation  = req.query.continuationToken || null;
+  let limit         = parseInt(req.query.limit || 1000);
   // TODO: Add support querying using prefix
```
Comment (gregarndt) [inline routes/artifacts.js]: Not something that changed in this PR, but just something I noticed is that we do no error checking here to ensure the task ID and run ID exist, but we do for the "/task/:taskId/artifacts" endpoint.

File: schemas/list-artifacts-response.yml [hunk 2]
```diff
@@ -52,6 +52,17 @@ properties:
...
+  continuationToken:
+    type:             string
+    title:            "Continuation Token"
+    description: |
+      Opaque `continuationToken` to be given as query-string option to get the
+      next set of artifacts.
+      This property is only present if another request is necessary to fetch all
+      results. In practice the next request with a `continuationToken` may not
+      return additional results, but it can. Thus, you can only be sure to have
+      all the results if you've callwith `continuationToken` until you get a
```
Comment (gregarndt) [inline schemas/list-artifacts-response.yml]: "callwith"?

File: test/api/artifact_test.js [hunk 3]
```diff
@@ -763,4 +763,77 @@ suite('Artifacts', function() {
...
+    assert(r2.continuationToken, "missing continuationToken");
+
+    debug("### listArtifacts, w. continuationToken");
+    let r3 = await helper.queue.listArtifacts(taskId, 0, {
+      continuationToken: r2.continuationToken,
+    });
+    assume(r3.artifacts.length).equals(1);
+    assume(r3.artifacts[0].contentType).equals('application/json');
+    assert(!r3.continuationToken, "unexpected continuationToken");
+    assume(r3.artifacts[0].name).not.equals(r2.artifacts[0].name);
```
Comment (gregarndt) [inline test/api/artifact_test.js]: Nice, I like that we check this!

**Comment by gregarndt** [comment]
:+1:

File: routes/artifacts.js (hunk 1 above)
Comment (jonasfj) [inline routes/artifacts.js]: @gregarndt, I guess we could check it... just because we want to return 404 rather than an empty list, right?

Comment (gregarndt) [inline routes/artifacts.js]: I was just mainly curious why we were checking it elsewhere but not here.  Nothing changed in this PR in regards to that so if we wanted to leave it unfixed or fix it in another PR at another time (just enter a bug?) then that's cool too.

Comment (jonasfj) [inline routes/artifacts.js]: just fixed it :)

**Pull request merged**